x-hooks/use-filters.tsx

```diff
@@ -122,6 +130,19 @@ export default function useFilters(): [
     updateRouterQuery();
   }
 
+  function checkOption({ repository, state, time }: updateMultipleProps) {
```

Comment:
this function will "always" change three states, which it shouldn't: it should update the state that its supposed to update; If user clicks on a repository filter, that's the only state that should change - so we don't trigger 3 changes when we are reading the states